__tests__/base.js

```diff
@@ -923,16 +946,26 @@ function runBaseTest(name, useProxies, autoFreeze, useListener) {
 					value: 1,
 					enumerable: false
 				})
+				// Non-enumerable primitive property that don't refer
+				Object.defineProperty(baseState, "baz", {
+					value: 1,
+					enumerable: false
+				})
+
+				// When using Proxy, even non-enumerable keys will be copied if it's changed.
+				const canReferNonEnumerableProperty = useProxies || useStrictShallowCopy
 				const nextState = produce(baseState, s => {
-					expect(s.foo).toBeTruthy()
-					expect(isEnumerable(s, "foo")).toBeFalsy()
-					s.bar++
-					expect(isEnumerable(s, "foo")).toBeFalsy()
-					s.foo.a++
-					expect(isEnumerable(s, "foo")).toBeFalsy()
-				})
-				expect(nextState.foo).toBeTruthy()
-				expect(isEnumerable(nextState, "foo")).toBeFalsy()
+					if (canReferNonEnumerableProperty) expect(s.foo).toBeTruthy()
+					if (useStrictShallowCopy) expect(isEnumerable(s, "foo")).toBeFalsy()
+					if (canReferNonEnumerableProperty) s.bar++
+					if (useStrictShallowCopy) expect(isEnumerable(s, "foo")).toBeFalsy()
+					if (canReferNonEnumerableProperty) s.foo.a++
+					if (useStrictShallowCopy) expect(isEnumerable(s, "foo")).toBeFalsy()
+				})
+				if (canReferNonEnumerableProperty) expect(nextState.foo).toBeTruthy()
+				if (useStrictShallowCopy)
+					expect(isEnumerable(nextState, "foo")).toBeFalsy()
+				if (useStrictShallowCopy) expect(nextState.baz).toBeTruthy()
```

Comment:
This test is ugly, but after some consideration, I think it should be the spec.

First, I thought that non-enumerable properties should not be visible from drafts when useStrictShallowCopy is false.

However, achieving this behavior requires checking whether the referenced property is non-enumerable, which introduces new overhead.